### PR TITLE
Model creation

### PR DESCRIPTION
**Pull request by faisalakb**
In this project apply the following features :  :100: 

Generate models against the tables created previously :+1: 
Insert the records into the tables  :+1: 

Create custom methods :+1: 

    Users
        A method that returns the 3 most recent posts for a given user.
    Posts
        A method that updates the posts counter for a user.
        A method which returns the 5 most recent comments for a given post.
    Comments
        A method that updates the comments counter for a post.
    Likes
        A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_post_counter
+    author.update(posts_counter: author.posts.count)
+  end
+
```
Comment (DeliceLydia) [inline app/models/post.rb:9]: ## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_post_counter` method private

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  def update_like_counter
+    post.update(likes_counter: post.likes.count)
+  end
```
Comment (DeliceLydia) [inline app/models/like.rb:7]: ## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `#update_like_counter` method every time a new like has been saved to the database, then make the `update_like_counter` method private

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :post, class_name: 'Post', foreign_key: :post_id
+  belongs_to :user, class_name: 'User', foreign_key: :user_id
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (DeliceLydia) [inline app/models/comment.rb:7]: ## Optional
- [ ] I suggest that you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the `#update_comments_counter` method every time a new comment has been saved to the database, then make the `update_comments_counter` method private

**Review by DeliceLydia — APPROVED** [review]
## Hi @faisalakb, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- The associations between models are set up correctly
- The User model has method that returns 3 most recent posts
- The Post model has a method that returns the 5 most recent comments


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by faisalakb** [comment]
thanks for the time and reviewer

**Pull request merged**